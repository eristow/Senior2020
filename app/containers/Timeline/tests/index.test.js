/**
 *
 * Tests for Timeline
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import configureStore from '../../../configureStore';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import { Timeline } from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';

describe('<Timeline />', () => {
  let store;

  beforeAll(() => {
    store = configureStore();
  });

  // TODO: not using this file, so test is commented out
  it.skip('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Timeline dispatch={dispatch} />
        </IntlProvider>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  // TODO: not using this file, so test is commented out
  it.skip('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Timeline />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
