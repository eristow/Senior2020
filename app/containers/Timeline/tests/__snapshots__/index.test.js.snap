// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timeline /> Should render and match the snapshot 1`] = `
<div>
  <h1
    class="H1-sc-3vkdoq-0 kuxxbb"
  >
    <span>
      Timeline
    </span>
  </h1>
  <div>
    <p
      class="P-sc-1c3t6r-0 jiFTib"
    >
      Volume Slider
    </p>
    <div
      style="display: inline;"
    >
      <div
        class="rc-slider"
        style="width: 200px; margin: 10px;"
      >
        <div
          class="rc-slider-rail"
        />
        <div
          class="rc-slider-track"
          style="left: 0%; width: 0%;"
        />
        <div
          class="rc-slider-step"
        />
        <div
          aria-disabled="false"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
          class="rc-slider-handle"
          role="slider"
          style="left: 0%; transform: translateX(-50%);"
          tabindex="0"
        />
        <div
          class="rc-slider-mark"
        />
      </div>
    </div>
    <p
      class="P-sc-1c3t6r-0 jiFTib"
    >
      Volume Slider With Tooltip
    </p>
    <div
      style="display: inline;"
    >
      <div
        class="rc-slider"
        style="width: 200px; margin: 10px;"
      >
        <div
          class="rc-slider-rail"
        />
        <div
          class="rc-slider-track"
          style="left: 0%; width: 0%;"
        />
        <div
          class="rc-slider-step"
        />
        <div
          aria-disabled="false"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
          class="rc-slider-handle"
          role="slider"
          style="left: 0%; transform: translateX(-50%);"
          tabindex="0"
        />
        <div
          class="rc-slider-mark"
        />
      </div>
    </div>
    <p
      class="P-sc-1c3t6r-0 jiFTib"
    >
      Slider With Fixed Values
    </p>
    <div
      style="display: inline;"
    >
      <div
        class="rc-slider"
        style="width: 200px; margin: 10px;"
      >
        <div
          class="rc-slider-rail"
        />
        <div
          class="rc-slider-track"
          style="left: 0%; width: 0%;"
        />
        <div
          class="rc-slider-step"
        />
        <div
          aria-disabled="false"
          aria-valuemax="10"
          aria-valuemin="0"
          aria-valuenow="0"
          class="rc-slider-handle"
          role="slider"
          style="left: 0%; transform: translateX(-50%);"
          tabindex="0"
        />
        <div
          class="rc-slider-mark"
        />
      </div>
    </div>
    <p
      class="P-sc-1c3t6r-0 jiFTib"
    >
      Number Input
    </p>
    <input
      class="InputNumber-fuyjcw-0 iCGUAJ"
      placeholder="0"
      type="number"
      width="5em"
    />
    <br />
    <p
      class="P-sc-1c3t6r-0 jiFTib"
    >
      Dropdown/select
    </p>
    <select
      class="Dropdown-sc-1lo2juv-0 gWmuJI"
      width="10em"
    >
      <option
        disabled=""
        value="1"
      >
        Select
      </option>
      <option
        value="2"
      >
        Test
      </option>
      <option
        value="3"
      >
        Test2
      </option>
    </select>
    <br />
    <p
      class="P-sc-1c3t6r-0 jiFTib"
    >
      Button
    </p>
    <div
      style="display: inline;"
    >
      <button
        class="StyledButton-sc-1vlwaww-0 cxCfee"
        width="4em"
      >
        Play
      </button>
    </div>
    <div>
      <input
        class="StyledCheckbox-sc-1n8d2c3-0 nyjIO"
        type="checkbox"
      />
      <p
        class="P-sc-1c3t6r-0 jiFTib"
      >
        Checkbox
      </p>
    </div>
    <p
      class="P-sc-1c3t6r-0 jiFTib"
    >
      Timeline button to select when drums play
    </p>
    <br />
    <p
      class="P-sc-1c3t6r-0 jiFTib"
    >
      Timeline Indicator
    </p>
  </div>
</div>
`;
